presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -490,7 +490,7 @@ private void invokeNoMoreSplitsIfNecessary()
 
         // S3 Select pushdown works at the granularity of individual S3 objects,
         // therefore we must not split files when it is enabled.
-        boolean splittable = getHeaderCount(schema) == 0 && getFooterCount(schema) == 0 && !s3SelectPushdownEnabled;
+        boolean splittable = !s3SelectPushdownEnabled && getFooterCount(schema) == 0 && getHeaderCount(schema) <= 1; // single header line files are still splittable
```

Comment:
Move the comment below, or restructure like this:

```
boolean splittable = true;
if (s3SelectPushdownEnabled) {
    // S3 Select pushdown works at the granularity of individual S3 objects
    splittable = false;
}
if (getFooterCount(schema) != 0 || (getHeaderCount(schema) != 0 && getHeaderCount(schema) != 1)) {
    // Files with skip.header.line.count/skip.foter.line.count are not splittable, except for skip.header.line.count = 1
    // In that case, second split will not try to process header line as data anyway.
    splittable = false;
} 
```

Reply:
Update `io.prestosql.plugin.hive.TestBackgroundHiveSplitLoader#testCsv`

Reply:
Good catch, thanks.